engine/src/flutter/display_list/utils/dl_matrix_clip_tracker.cc

```diff
@@ -97,6 +97,37 @@ void DisplayListMatrixClipState::clipRRect(const DlRoundRect& rrect,
   }
 }
 
+void DisplayListMatrixClipState::clipRSuperellipse(
+    const DlRoundSuperellipse& rse,
+    DlClipOp op,
+    bool is_aa) {
+  DlRect bounds = rse.GetBounds();
+  if (rse.IsRect()) {
+    return clipRect(bounds, op, is_aa);
+  }
+  switch (op) {
+    case DlClipOp::kIntersect:
+      adjustCullRect(bounds, op, is_aa);
+      break;
+    case DlClipOp::kDifference: {
+      if (rsuperellipse_covers_cull(rse)) {
+        cull_rect_ = DlRect();
+        return;
+      }
+      auto radii = rse.GetRadii();
+      DlRect safe = bounds.Expand(
+          -std::max(radii.top_left.width, radii.bottom_left.width), 0,
+          -std::max(radii.top_right.width, radii.bottom_right.width), 0);
+      adjustCullRect(safe, op, is_aa);
+      safe = bounds.Expand(
+          0, -std::max(radii.top_left.height, radii.top_right.height),  //
+          0, -std::max(radii.bottom_left.height, radii.bottom_right.height));
+      adjustCullRect(safe, op, is_aa);
+      break;
+    }
```

Comment:
After looking at this method, I'm wondering if it is worth having a common rect/radii helper-ish method that both RRect and RSuperEllipse defer to. Is there any difference in the logic?

Reply:
No, they're exactly the same. I've extracted the logic as `RoundingRadiiSafeRects`.